azure-iot-device/tools/create_x509_chain.py

```diff
@@ -0,0 +1,190 @@
+import os
+import re
+import argparse
+import getpass
+
+
+def create_custom_config():
+    # This is very specific to installations in various system
+    openssl_path = os.getenv("OpenSSLDir")
+    config_path = openssl_path + "/bin/openssl.cfg"
+    with open(config_path, "r") as openssl_config:
+        config = openssl_config.read()
+    lines = config.splitlines()
+    policy_loose_found = False
+    policy_any_found = False
+
+    policy_any_regex = re.compile(r"\s*\[\s*policy_anything\s*\]\s*")
+
+    # First, try to find policy_anything in the openssl config file
+    for number, line in enumerate(lines):
+        if re.search(policy_any_regex, line):
+            policy_any_found = True
+            break
+    # Of not found the try the search with policy_loose
+    if not policy_any_found:
+        policy_loose_regex = re.compile(r"\s*\[\s*policy_loose\s*\]\s*")
+        for number, line in enumerate(lines):
+            if re.search(policy_loose_regex, line):
+                policy_loose_found = True
+                break
+
+    list_of_lines = list()
+    ca_default_regex = re.compile(r"\s*\[\s*CA_default\s*\]\s*")
+    ca_default_section_found = False
+    policy_regex = re.compile(r"\s*\s*(policy\s*=\s*policy_strict|policy\s*=\s*policy_match)")
+    with open("C:/OpenSSL-Win64/bin/openssl.cfg", "r") as change_openssl_config:
```

Comment:
>C:/OpenSSL-Win64/bin/openssl.cfg [](start = 15, length = 32)

Only valid on Windows, and with 64 bit at that.

Reply:
changed to environment variable like the previous one. 

---
In reply to: [297798367](https://github.com/Azure/azure-iot-sdk-python-preview/pull/116#discussion_r297798367) [](ancestors = 297798367)